modules/irsa/main.tf

```
@@ -13,6 +13,22 @@ resource "kubernetes_namespace_v1" "irsa" {
   }
 }
 
+resource "kubernetes_secret_v1" "irsa" {
+  count = var.create_kubernetes_service_account ? 1 : 0
+  metadata {
+    name        = format("%s-token-secret", var.kubernetes_service_account)
+    namespace   = try(kubernetes_namespace_v1.irsa[0].metadata[0].name, var.kubernetes_namespace)
+    annotations = {
+      "kubernetes.io/service-account.name"      = var.kubernetes_service_account
```

Change this and then drop the `depends_on`

```suggestion
      "kubernetes.io/service-account.name"      = try(kubernetes_service_account_v1.[0].metadata[0].name, var.kubernetes_service_account)
```